Modules/_bz2module.c

```diff
@@ -684,9 +660,6 @@ BZ2Decompressor_dealloc(PyObject *op)
     }
     BZ2_bzDecompressEnd(&self->bzs);
     Py_CLEAR(self->unused_data);
-    if (self->lock != NULL) {
```

Comment:
In the zstd code, we have a
`assert(!PyMutex_IsLocked(&self->lock));` to ensure that the locking is handled correctly. I think it would be good to add that here too.